src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -7,50 +7,158 @@
+        try {
+            return of(id);
+        } catch (IOException ex) { // The error is not a hazard in CheckForNull logic
+            LOGGER.log(Level.WARNING, "Cannot retrieve a fingerprint for Docker id="+id, ex);
+        }
+        return null;
+    }
+    
+    /**
+     * Get or create a {@link Fingerprint} for the image.
+     * @param run Origin of the fingerprint (if available)
+     * @param id Image ID. Only 64-char full IDs are supported.
+     * @return Fingerprint for the specified ID
+     * @throws IOException Fingerprint load/save error
+     */
+    public static @Nonnull Fingerprint makeForImage(@CheckForNull Run<?,?> run, @Nonnull String id) throws IOException {
+        return Jenkins.getInstance().getFingerprintMap().getOrCreate(run, "<docker-image>", getFingerprintHash(id));
+    }
+    
+    /**
+     * Get or create a {@link Fingerprint} for the container.
+     * @param run Origin of the fingerprint (if available)
+     * @param id Image ID. Only 64-char full IDs are supported.
+     * @return Fingerprint for the specified ID
+     * @throws IOException Fingerprint load/save error
+     */
+    public static @Nonnull Fingerprint makeForContainer(@CheckForNull Run<?,?> run, @Nonnull String id) throws IOException {
+        return Jenkins.getInstance().getFingerprintMap().getOrCreate(run, "<docker-container>", getFingerprintHash(id));
     }
 
     /**
+     * Retrieves a facet from the {@link Fingerprint}.
+     * The method suppresses {@link IOException} if a fingerprint loading fails.
+     * @param <TFacet> Facet type to be retrieved
+     * @param id Docker item ID. Only 64-char full IDs are supported
+     * @param facetClass Class to be retrieved
+     * @return First matching facet. Null may be returned if the loading fails
+     */
+    public static @CheckForNull @SuppressWarnings("unchecked")
+            <TFacet extends FingerprintFacet> TFacet getFacet
+            (@Nonnull String id, @Nonnull Class<TFacet> facetClass) { 
+        final Fingerprint fp = ofNoException(id); 
+        return (fp != null) ? getFacet(fp, facetClass) : null;
+    }
+            
+    /**
+     * Retrieves a facet from the {@link Fingerprint}.
+     * The method suppresses {@link IOException} if a fingerprint loading fails.
+     * @param <TFacet> Facet type to be retrieved
+     * @param id Docker item ID. Only 64-char full IDs are supported
+     * @param facetClass Class to be retrieved
+     * @return First matching facet. Null may be returned if the loading fails
+     */
+    @SuppressWarnings("unchecked")
+    public static @Nonnull <TFacet extends FingerprintFacet> Collection<TFacet> getFacets
+            (@Nonnull String id, @Nonnull Class<TFacet> facetClass) { 
+        final Fingerprint fp = ofNoException(id);
+        return (fp != null) ? getFacets(fp, facetClass) : CollectionUtils.EMPTY_COLLECTION;
+    }        
+    
+    //TODO: deprecate and use the core's method when it's available
```

Comment:
But it looks like it is only used internally in this class, right? So we could avoid polluting the `docker-commons` API with it?